osquery/tables/system/darwin/system_info.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <osquery/database.h>
+#include <osquery/tables.h>
+#include "osquery/core/conversions.h"
+
+#include <CoreFoundation/CoreFoundation.h>
+#include <IOKit/IOKitLib.h>
+
+namespace osquery {
+namespace tables {
+
+Status getCpuId(std::string& cpu_id) {
+  auto matching = IOServiceMatching("IOPlatformExpertDevice");
+  if (matching == nullptr) {
+    return Status(1, "Could not get service matching IOPlatformExpertDevice");
+  }
+
+  io_iterator_t it;
+  auto kr = IOServiceGetMatchingServices(kIOMasterPortDefault, matching, &it);
+  if (kr != KERN_SUCCESS) {
+    return Status(1, "Could not get iterator");
+  }
+
+  // There should be only one service, so just grab the first one
+  io_service_t service;
+  service = IOIteratorNext(it);
+  if (service == 0) {
+    return Status(1, "Could not iterate to get service");
+  }
+
+  CFStringRef serialNumber = (CFStringRef)IORegistryEntryCreateCFProperty(
+    service,
+    CFSTR("IOPlatformSerialNumber"),
+    kCFAllocatorDefault,
+    kNilOptions);
+  IOObjectRelease(service);
+  if (serialNumber == nullptr) {
+    return Status(1, "Could not read serial number property");
+  }
+
+  cpu_id = stringFromCFString(serialNumber);
+  CFRelease(serialNumber);
+  if (cpu_id.empty()) {
+    return Status(1, "cpu_id was empty");
+  }
+
+  return Status(0, "OK");
+}
+
+QueryData genSystemInfo(QueryContext& context) {
+
+  Row r;
+  r["hostname"] = TEXT(osquery::getHostname());
+
+  std::string hostuuid;
+  auto status = osquery::getHostUUID(hostuuid);
+  if (!status.ok()) {
+    hostuuid = "<unknown>";
```

Comment:
This is more of a question to @marpaia / @theopolis:
Usually when we have unknown values for any column, we have have left it to its default null value. See `hardware_events` or `pci_devices` where sometimes `model`, `serial`, `vendor` are unknown. 

This makes it easy to use SQL comparators for example: `select * from hardware_events where serial != ""` or `where foo != -1`. I think we should stick to defaulting null values. Thoughts?